SuiteCRM/SuiteCRM
# Feature: Ability to add custom code to bootup

## Description
The ability to add customisations in 2 places: in the beginning after the config is loaded and once the user is loaded.

## Motivation and Context
1. We need to execute some code after the config is loaded in order to get better reporting from our application performance monitoring system, so following the `custom/include` pattern: if
 `custom/include/entryPoint.php` exists it'll be `required` .
2. Implements a `after_load_user` hook so custom code can be executed on the loaded user before continuing with the application logic, this is different from the `after_login` as that only fires once when the user performs the login, where this will fire for every request when the user is retrieved from the session

## How To Test This
Both need to be tested manually as there are no unit tests available for the `entryPoint.php` and `SugarApplicationTest::testloadUser` is commented out due to `die` logic...
1. Add custom code in `custom/include/entryPoint.php` and make sure it gets executed.
2. Adding an application hook registration for `after_load_user` and ensuring the function is executed
```php
if (!isset($hook_array['after_load_user'])) {
    $hook_array['after_load_user'] = [];
}
$hook_array['after_load_user'][] = [
    1,
    'Check User',
    'custom/<HookClass>.php',
    '<HookClass>',
    '<checkUserFunction>'
];
```
I've manually tested these both with the file and hook present and not present.

## Types of changes

### Final checklist
